### PR TITLE
CI: simplify self-hosted check

### PR DESCRIPTION
**Pull request by jschwe**
See also the official documentation on the
runner context:

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#runner-context

> The environment of the runner executing the job. Possible values are: `github-hosted` for GitHub-hosted runners provided by GitHub, and `self-hosted` for self-hosted runners configured by the repository owner.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

### CODE REVIEW

**Comment by mukilan** [comment]
The [main PR build](https://github.com/servo/servo/actions/runs/12388979528) has failed due to a syntax error:

```

Invalid workflow file: .github/workflows/main.yml#L28
The workflow is not valid. In servo/servo/.github/workflows/dispatch-workflow.yml@8f50ec8ef34b987d9cce07bb14fd5ed7f46dc1bc (Line: 28, Col: 11): Error from called workflow servo/servo/.github/workflows/windows.yml@8f50ec8ef34b987d9cce07bb14fd5ed7f46dc1bc (Line: 72, Col: 23): Unrecognized named-value: 'runner'. Located at position 1 within expression: runner.environment == 'self-hosted'
```

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -98,36 +98,36 @@ jobs:
     secrets: inherit
     with:
       unique-id: ${{ needs.runner-select.outputs.unique-id }}
-      is-self-hosted: ${{ fromJSON(needs.runner-select.outputs.is-self-hosted) }}
+      is-self-hosted: ${{ runner.environment == 'self-hosted' }}
```
Comment (mukilan) [inline .github/workflows/linux.yml:101]: Is this change correct? Won't the runner-timeout job itself be ran on github hosted runners?

Comment (jschwe) [reply .github/workflows/linux.yml]: Oh, thanks for catching that! I was running search and replace and apparently didn't pay enough attention to the context :(

**Review by mukilan — APPROVED** [review]
Thank you!

**Review by delan — APPROVED** [review]

**Pull request merged**